### PR TITLE
Editor / CPT: Use type label for saved post view button

### PR DESCRIPTION
**Pull request by aduth**
Related: #3695

This pull request seeks to use a post type's `labels.view_item` string in rendering the action bar View button tooltip text. This furthers our focus on custom post types, as the current editor only supports from hard-coded post types (page and post).

Further, it also resolves a styling issue where the View button is misaligned with other buttons in the action bar. It does so by refactoring the link to be displayed as a `<Button borderless />` component and applying consolidated styling to all buttons in the action bar.

Before|After
---|---
Before|After

__Testing instructions:__

Verifying custom post type labels is currently blocked by #4193. In the interim, test to ensure that "View Post" and "View Page" labels display correctly for saved posts and pages, respectively. Also verify that alignment of buttons is corrected, while still preserving targeted styling (e.g. red hover on delete).

### CODE REVIEW

File: client/post-editor/editor-action-bar/view-label.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	const post = getEditedPost( state, siteId, getEditorPostId( state ) );
+	if ( ! post ) {
+		return props;
+	}
+
+	return Object.assign( props, {
+		typeSlug: post.type,
+		type: getPostType( state, siteId, post.type )
+	} );
+} )( localize( EditorActionBarViewLabel ) );
```
Comment (gwwar) [inline client/post-editor/editor-action-bar/view-label.jsx]: Fun :smile:

**Comment by gwwar** [comment]
Heads up on the more generic fix IE11 fix: #4405

**Comment by gwwar** [comment]
:+1: Code changes look good here. Feel free to merge or ping me to test the full path when you rebase. IE11 displays the huge icons, but it should be fixed once you pull in changes from master.

**Comment by aduth** [comment]
Thanks for the review @gwwar .

Verified that icons display correctly in IE11 after rebase:

image

File: client/post-editor/editor-action-bar/view-label.jsx (hunk 1 above)
Comment (aduth) [inline client/post-editor/editor-action-bar/view-label.jsx]: >Fun :smile: 

Who ever said custom post type support would be easy? :laughing:

**Pull request merged**